spectrochempy/analysis/pls.py

```diff
@@ -0,0 +1,317 @@
+# -*- coding: utf-8 -*-
+# ======================================================================================
+# Copyright (©) 2015-2023 LCS - Laboratoire Catalyse et Spectrochimie, Caen, France.
+# CeCILL-B FREE SOFTWARE LICENSE AGREEMENT
+# See full LICENSE agreement in the root directory.
+# ======================================================================================
+"""
+Implementation of Partial Least Square regression (using scikit-learn library)
+"""
+
+import traitlets as tr
+from sklearn import cross_decomposition
+
+from spectrochempy.analysis._base import (
+    CrossDecompositionAnalysis,
+    _wrap_ndarray_output_to_nddataset,
+)
+from spectrochempy.core.dataset.nddataset import NDDataset
+from spectrochempy.utils.decorators import signature_has_configurable_traits
+from spectrochempy.utils.docstrings import _docstring
+
+__all__ = ["PLS"]
+__configurables__ = ["PLS"]
+
+
+# ======================================================================================
+# class PLS
+# ======================================================================================
+_docstring.delete_params("AnalysisConfigurable.parameters", "copy")
+
+
+@signature_has_configurable_traits
+class PLS(CrossDecompositionAnalysis):
+    _docstring.delete_params("DecompositionAnalysis.see_also", "PLS")
+
+    __doc__ = _docstring.dedent(
+        """
+    Partial Least Squares regression (PLS).
+
+    The  Partial Least Squares regression wraps the
+    `sklearn.cross_decomposition.PLSRegression` model, with few
+    additional methods.
+
+    Parameters
+    ----------
+    %(AnalysisConfigurable.parameters.no_copy)s
+
+    See Also
+    --------
+    %(DecompositionAnalysis.see_also.no_PLS)s
```

Comment:
Do we really want to refer to decomposition methods here?
If yes we should also add PLS in See Also of the other method?

Reply:
Agree. I remove it